### DISCUSSION TITLE
Allow `SELECT many` and `INSERT` on SQL tables without a primary key

### DISCUSSION THREAD

**Discussion by ItsMeBrianD** [Budibase/budibase#3840] [Ideas]
Hey all,

As the title says, I'm unable to link my existing PostgreSQL database to a new budibase instance because I receive the error message `No primary key constraint found for the following`.

While one fix seems fairly straightforward (adding a primary key to every table), this doesn't seem like a restriction that needs to be in place. Any ideas on how to fix / circumvent this?

**Comment by mjashanks** [comment]
The reason that we require a primary key is because we support CRUD. So, in order to automate `UPDATE one` and `SELECT one`, we need to use a `WHERE primary_key = value` clause.

You can still write a custom query if you need to get around this. 

Would it be annoying to have to write a custom query for what you are doing? I'd be keen to learn more about what your needs are here.

**Reply by ItsMeBrianD** [reply]
While custom queries are an option (and something I'm playing with right now); not having implicit access to the entire database (~30 tables) is a little disheartening.

Right now I'm just playing around with budibase a little bit to see if it will be a good fit for building tools / reports / data vis for an esports community I am help run

**Reply by mjashanks** [reply]
This has come up a few times. 

It would be good to only allow `SELECT` and `INSERT` for tables without a primary key. However, this means that tables would have a "Feature" set - which is work for us to support.

I'll turn this into an idea anyway, so that it can be considered in future roadmap discussions.

**Reply by ItsMeBrianD** [reply]
That would be great, most of the data is going to be read only anyways so being able to leverage the built-in stuff for display would be great.

As a workaround adding primary keys should be doable

**Reply by ItsMeBrianD** [reply]
Another thing that would be great would be importing only the tables that have primary keys (i.e. leaving out those that don't have them) as an easier way to have partial support
